Fix Copy to clipboard button in SFT

naotone opened the pull request:
Issue #1137

Following implementations were added:
- Display an error message when clipboard copying fails.
- Automatically select the text after an error occurs.

This error was caused by Secure Contexts. The navigator.clipboard.writeText function can only be executed in environments such as HTTPS or localhost. Domains like http://example.local do not support clipboard copying.
https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts

document.execCommand('copy') could be used as an alternative, it was not added since it is deprecated.
The Final Result data in the screenshot was created by me for testing.

![20250301-003502@2x](https://github.com/user-attachments/assets/0ca71ff8-c0f5-4706-b090-ae65da927e96)


<!--
Thank you for contributing to TensorZero!

Before submitting your PR, make sure you've read **[Contributing to TensorZero](https://github.com/tensorzero/tensorzero/blob/main/CONTRIBUTING.md)**.

In particular, make sure you've run `pre-commit` and any tests relevant to your changes (including E2E tests for changes to the gateway).

By submitting this PR, unless otherwise specified, you agree to license your contributions under the **[Apache 2.0 license](https://github.com/tensorzero/tensorzero/blob/main/LICENSE)**.
-->

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Fixes clipboard copy error handling in `SFTResult.tsx`, displaying an error message and selecting text on failure.
> 
>   - **Behavior**:
>     - Adds error handling in `handleCopy` function in `SFTResult.tsx` for clipboard copy failures.
>     - Displays error message and selects text if clipboard copy fails.
>     - Uses `useRef` to manage text selection in `Textarea`.
>   - **UI**:
>     - Updates `Button` to use `handleCopy` for clipboard operations.
>     - Displays error message below `Button` if copy fails.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=tensorzero%2Ftensorzero&utm_source=github&utm_medium=referral)<sup> for 4fa895547e4a0a822e874eaa6aeb51cfa21886bd. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

virajmehta commented:
@naotone thank you for the contribution! I'm not an expert on browser features so when I built this I tested locally and thought it was good enough. I think the best UX here is just to not show the button when it's not supported. I think this is possible with [this](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts#feature_detection) API:

```
if (window.isSecureContext) {
  // Page is a secure context so service workers are now available
  navigator.serviceWorker.register("/offline-worker.js").then(() => {
    // …
  });
}
```
we could probably conditionally render the button only under those circumstances. Mind making that change? Thanks

naotone commented:
@virajmehta Thank you for your feedback! Auto-selecting is intended to avoid manual full selection on touch devices such as tablets and mobile phones. However, since TensorZero is primarily a desktop platform, it might be fine to hide it in this case.

In the latest commit, I’ve adjusted the logic so that the copy button is only displayed when window.isSecureContext and navigator.clipboard are both available. Additionally, I’ve introduced a forceShowCopyButton prop (default: false) in case there’s a need to override this behavior.
If you’d prefer a simpler approach where the button is always hidden when unsupported, let me know, and I can make that change. Thanks!

virajmehta commented:
@naotone this seems like a sensible approach to me! I ran a job myself in a secure & insecure context to test. 

Thank you for the contribution!
